frontend/server/src/DAO/ACLs.php

```diff
@@ -10,4 +10,43 @@
  * @access public
  */
 class ACLs extends \OmegaUp\DAO\Base\ACLs {
+        /**
+     * Get all ACLs owned by a user along with their type, alias, and an empty users array.
+     *
+     * @param int $userId The user ID whose ACLs are being fetched.
+     * @return list<array{acl_id: int, type: string, alias: string, users: list<array<string, never>>}>
+     */
+    public static function getUserOwnedAclTypesWithAliases(int $userId): array {
+        $sql = '
+            SELECT a.acl_id, c.alias, "contest" AS type
+            FROM ACLs a
+            INNER JOIN Contests c ON c.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+            UNION
+            SELECT a.acl_id, c.alias, "course" AS type
+            FROM ACLs a
+            INNER JOIN Courses c ON c.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+            UNION
+            SELECT a.acl_id, p.alias, "problem" AS type
+            FROM ACLs a
+            INNER JOIN Problems p ON p.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+            UNION
+            SELECT a.acl_id, g.alias, "group" AS type
+            FROM ACLs a
+            INNER JOIN Groups_ g ON g.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+        ';
+        $params = [$userId, $userId, $userId, $userId];
```

Comment:
If you want to avoid repeating the same variable multiple times:
```suggestion
        $params = array_fill(0, 4, $userId);
```